### PR TITLE
Add Github issue template for new releases and update release tooling

### PR DESCRIPTION
**Pull request by hoffie**
This PR adds the release checklist as an issue template so that we can easily create new Issues based on that.

The checklist is a slightly improved version of what we had for 3.7.0.

I'm still planning to perform some updates and possibly add a template for translations as well. I'm therefore setting this to Draft status right now. Feel free to provide feedback regarding the existing content or the process itself though.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:24]: I think there's a missing step we talked about: Open Issues for each language (so that we can track the status of translation, and so that translators can discuss issues if they want before they submit PRs). Assigning these to their respective translators also implies we need a list of who is volunteering to which language. Right now we have no Germans, for example. Translators then create PRs for both app and website if needed, linking them to their Issue.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:25]:
```suggestion
  - [ ] Ping translators. Translators should open PRs with their changes against the `translate3_y_z` branch (?),  reference this issue ("Add German translations for 3.y.z") and link to their respective Issue.
```

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:26]: Should we agree what happens if the PRs don't come in by the target date? For the web site, do we delete any affected .md files in the wiki/[lang]/ directory so that out-of-date translations don't exist? The language switcher link will still be there but just kicks back to the English. Not sure what to do for the app.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:29]: Would this have been done as part of the previous "ping" (ie setting up the Issue for them to link their PRs to)?

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:37]: Again, by target date? And what to do if nothing comes in.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:41]: Oh I thought we'd have done this earlier since app versions are "tied" to website releases.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:55]: Is this a separate announcement? If so, I guess we un-pin the beta? Also don't forget to turn comments off and pin it.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [inline .github/ISSUE_TEMPLATE/release_new_version.md:69]:
```suggestion
- [ ] Check that all Issues and PRs tagged for this release are in Done/Closed state. Create a milestone for the next minor release 
```

**Comment by gilgongo** [comment]
BTW turning comments off on the release announcement is a learning from SourceForge, as such announcements would sometimes turn into mini-forum of random tech support requests, complaints, "thanks" and other things that should have been opened as separate discussions.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [reply .github/ISSUE_TEMPLATE/release_new_version.md:16]: Also, perhaps we should encourage the shepherd to chase people to remind people to check the "waiting on team"  items that are also flagged for release?

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:20]: Hrm, looking at the previous [rc announcement](https://github.com/jamulussoftware/jamulus/discussions/1217), it looks like most (all?) of the replies for relevant to the release candidate. I'd rather tend to not lock these as it makes it easier to get feedback. I think something like "I have made a public testserver with 3.7.0rc1 in Düsseldorf, Genre Rock, named Testserver 3.7.0rc1." is easily posted to such a topic (and provides relevant information), but I suspect the author would not have gone the way to open a new discussion for that?

Comment (gilgongo) [reply .github/ISSUE_TEMPLATE/release_new_version.md:20]: See my comment below, but it's just better to have people post separate topics rather than use the release announcement as a sort of forum within a forum so that we can spot and manage things easier.

Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:20]: Yeah, I agree for the final release announcement which received too many unrelated replies. I'm mainly wondering if we should handle pre-releases differently and keep them unlocked.

**Comment by pljones** [comment]
> 
> 
> BTW turning off comments for release announcements is a learning from SourceForge, as such announcements would sometimes turn into mini-forum of random tech support requests, complaints, "thanks" and other things that should have been opened as separate discussions.

For the release, yes.  Not for nightlies, betas and candidates, though - it's better to keep the discussions on the same thread as the announcement for those, really.  You don't want the forum overrun with odd posts that may relate to an out of date RC being followed up a week later with a "So what's been done about it?" because no one spotted it - or because the raiser didn't see that it had been fixed in the new RC, etc.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [reply .github/ISSUE_TEMPLATE/release_new_version.md:55]: And yes, maybe comments on for interim releases then.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [reply .github/ISSUE_TEMPLATE/release_new_version.md:20]: Yes, interim releases before the final release announcement should have comments on in that case.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:16]: Thanks, will change to a filtered Project board link and add a *Waiting on team* reminder.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:26]: > Not sure what to do for the app.

Nothing. :)
Untranslated strings will simply appear in English and I think that's fine. If late translations come in, we could do a very quick .1 release.

I have no strong opinion on the website stuff. If the outdated translation becomes wrong (i.e. because it lacks a required compatibility update -- unlikely, I guess?) we should probably back it out. If it is simply missing some documentation about new features, it might be OK to keep it? Would be nice to have a banner like "This is outdated, see the English version for the latest updates"? Probably not trivial though. :(

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:55]: Yes, will be updated, thanks.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:69]: Thanks, next update will have this.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: Yes, I think App translations and App installer translations should now be handled with a single Issue per language. Installer translations will also be updated less often, I think. I have combined that now.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: I have added a note to communicate the deadline and a note that it is ok if a language is not fully translated for a release.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: I have split both *App translations* and *Website translations* into a *Prepare* (= Ping) and *Finish* (= Review/Merge) phase now to make the timing clearer.

**Comment by pljones** [comment]
Until other documentation changes intervene, we need a step here to check `./Jamulus -h` output against the `en-Command-Line-Options.md` wiki page.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:23]: I've added the information about individual Issues, which was missing.
I'm leaving this comment unresolved for now as we still need a way to find out who is responsible for each language. `git log $LANG*.md` may help if there's nothing else?

**Comment by hoffie** [comment]
> Until other documentation changes intervene, we need a step here to check `./Jamulus -h` output against the `en-Command-Line-Options.md` wiki page.

Good idea. Added that after code freeze and before starting translations.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (gilgongo) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: Yes it's hard to know what to do. People have a right to expect up to date translations or none at all really, since a message that what they're reading is out of date is almost worse! We did in fact take a decision to only translate the "core" pages for now. But that seems to have been sort of ignored though...

**Comment by hoffie** [comment]
I have updated the checklist in several places, especially around translations. I have also added automation for creating the translation tickets as part of this PR.

**Comment by hoffie** [comment]
Updated again. The script for generating the issues can now handle website translations as well.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:24]: This is now done via the new `tools/create-translation-issues.sh`. It contains a static list of translators though. Will need to update that based on `git log` if required (this is a checklist entry on its own).

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (pljones) [inline .github/ISSUE_TEMPLATE/release_new_version.md:60]: Smoke test should come before tagging as latest - i.e. off the r3_x_y build once it's been created.  Those are the "final" binaries and the smoke test should come as soon as possible - in case it's not actually final! :)

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (ann0see) [inline .github/ISSUE_TEMPLATE/release_new_version.md:37]: We didn't do this until now. The process was:

changes > translate (manual) > release (manual) > changes (via automation)

Any reasons for the change?

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (ann0see) [inline .github/ISSUE_TEMPLATE/release_new_version.md:67]: As above.

File: tools/create-translation-issues.sh
Comment (ann0see) [inline tools/create-translation-issues.sh:1]: Great work! I assume this will make the process a lot easier. Since there are many other release scripts too, shouldn't we put them in folder called /release under /tools/?

Comment (pljones) [reply tools/create-translation-issues.sh:1]: `tools` used to be about other repositories contributing to Jamulus.

Comment (hoffie) [reply tools/create-translation-issues.sh:1]: > Since there are many other release scripts too, shouldn't we put them in folder called /release under /tools/?

I was rather wondering if we should move such things to a separate repo as there is no need to distribute them as part of the releases. Sure, it's handy for us to have them available right next to the code, but moving them outside may also have benefits.

Not sure what the scope of the repo and as such what the name should be. Should it be limited to release `tools`? Or general `automation` maybe?

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:37]: It was just me misunderstanding the current process. :)
Will update the PR in the coming days.

File: tools/create-translation-issues.sh
Comment (ann0see) [reply tools/create-translation-issues.sh:1]: Probably automation or maintain? Not sure if tools is correct

Comment (hoffie) [reply tools/create-translation-issues.sh:1]: I suggest merging this PR with the create-translation-issues.sh script and move it afterwards along with the other scripts. I've opened #1764 to track this.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md:67]: Updated, thanks.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: Removed this step (and updated the later one).

**Comment by hoffie** [comment]
I think it makes sense to give this a final review and merge. The checklist started living within this PR which makes little sense. It's quite certain that we will have to update it again (@ignotus666's po4a change comes to mind), but we should do that in future PRs.

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (ann0see) [inline .github/ISSUE_TEMPLATE/release_new_version.md:66]:
```suggestion
- [ ] Publish Website release by merging `translate3_y_z` into `release`
```

**Review by ann0see — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release_new_version.md
Comment (hoffie) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: Thanks, I meant to write that but forgot that update. :/
Done now.

**Review by ann0see — APPROVED** [review]

**Comment by ann0see** [comment]
Something else could be added to the documentation workflow:

- [ ] Check for broken links with a link checker locally

**Comment by hoffie** [comment]
> Something else could be added to the documentation workflow:
> 
>     * [ ]  Check for broken links with a link checker locally

Good idea. Where would you add it? Probably before merging `translate3_y_z` to `release` in any case. Does it also make sense to run this before pinging translators? Is there a checking tool which you can suggest? For the jamulus.io monitoring I've used a `wget` call, but it cannot check anchors, IIRC.

**Comment by pljones** [comment]
> Does it also make sense to run this before pinging translators?

I'd say so -- in case the wording has to change as part of fixing the link.

**Comment by hoffie** [comment]
> > Does it also make sense to run this before pinging translators?
> 
> I'd say so -- in case the wording has to change as part of fixing the link.

Updated.

File: tools/create-translation-issues.sh
Comment (pljones) [reply tools/create-translation-issues.sh:1]: That, in itself, has benefits, as it implies it's more of an overview - covering all assets, not either jamulussoftware/jamulus or jamulussoftware/jamuluswebsite as would currently be the case.  Seems a positive step to me.

File: tools/create-translation-issues.sh
Comment (pljones) [inline tools/create-translation-issues.sh:17]: Should be `../` at the front of the example.

File: tools/create-translation-issues.sh
Comment (pljones) [inline tools/create-translation-issues.sh:64]: `- Update your fork to current ``master``,`
This should say `from jamulussoftware/jamulus master` or `from jamulussoftware/jamuluswebsite master` (depending...)

File: tools/create-translation-issues.sh
Comment (pljones) [inline tools/create-translation-issues.sh:99]: `- Start your work in your fork on a branch based on jamuluswebsite'"'"'s ``${TRANSLATE_BRANCH}`` branch.`

The code version is already clearer...  This needs to make it clear that local fork's master needs updating from jamulussoftware/jamuluswebsite master, then the local ``${TRANSLATE_BRANCH}`` branch updated, and then work started on a branch from the ``${TRANSLATE_BRANCH}`` branch.  If I've understood the process right...

File: tools/create-translation-issues.sh
Comment (pljones) [inline tools/create-translation-issues.sh:121]: Purely as a coding style point: this active section of code is sandwiched between two passive declarations.  It could be overlooked.  I'd move it to the top (just after the `set -eu`).

File: tools/create-translation-issues.sh
Comment (pljones) [inline tools/create-translation-issues.sh:1]: What a lovely bit of bash :)

File: tools/create-translation-issues.sh
Comment (hoffie) [reply tools/create-translation-issues.sh:17]: Yep, thanks, updated.

File: tools/create-translation-issues.sh
Comment (hoffie) [reply tools/create-translation-issues.sh:64]: Updated.

File: tools/create-translation-issues.sh
Comment (hoffie) [reply tools/create-translation-issues.sh:99]: There is no `master` in the website repo (`release` would be the most similar branch). However, `${TRANSLATE_BRANCH}` is the only branch which translators should have to care about (well, 3.8.0 was different for $REASONS...). This branch is a one-time branch (e.g. `translate3_8_0`), so there should be no need to update/rebase.
As the process is supposed to be reworked regarding tooling (po4a) and branching, I'd like to keep it like this for now.

(cc @ignotus666 @ann0see @gilgongo)

File: tools/create-translation-issues.sh
Comment (hoffie) [reply tools/create-translation-issues.sh:121]: Yep, good point, have moved it to the top (but kept the argument check / Usage information before it).

**Comment by ann0see** [comment]
There's still something for the website (delete translate branch)

**Comment by hoffie** [comment]
> There's still something for the website (delete translate branch)

Added now.

File: tools/create-translation-issues.sh
Comment (pljones) [reply tools/create-translation-issues.sh:105]: OK, if it's all going to change, no problem.  (However, make whatever's written clear enough that a first time translator will be able to read it and know what steps to follow.)

**Comment by ann0see** [comment]
@jamulussoftware/maindevelopers shouldn‘t this be Part of 3.8.1 too?

**Comment by pljones** [comment]
> @jamulussoftware/maindevelopers shouldn‘t this be Part of 3.8.1 too (at least the scripts)?

- is it what we're following for 3.8.1?
- if not, why not?  (i.e. what needs to change in what we're doing or what needs changing in this?)
- if so, yes and merge it.

**Comment by ann0see** [comment]
I've uploaded the issue template on the website, so this should be removed. Everything else should be ok?

**Pull request merged**
